### DISCUSSION TITLE
Scope for the new Flows system

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#15025] [Archive]
I have a flow with a manual trigger for a selected collection (Season):

<img width="860" alt="Screen Shot 2022-08-11 at 14 32 52" src="https://user-images.githubusercontent.com/1007051/184134060-975ea27b-09c2-4fcd-9174-289bbdcea2b4.png">

So I have this button in the item page:

<img width="358" alt="Screen Shot 2022-08-11 at 14 34 11" src="https://user-images.githubusercontent.com/1007051/184134375-72a03626-3af0-49ff-bc02-937bb83d4500.png">

When I click on the button, I want to get info from the other collections (Matchday & Matches), do some calculations and then update the Season data.

I can't do that with built-int operations, right?

But could I do it with a custom one?
https://docs.directus.io/extensions/operations/

**Comment by rijkvanzanten** [comment]
> When I click on the button, I want to get info from the other collections (Matchday & Matches), do some calculations and then update the Season data.
> I can't do that with built-int operations, right?

You should be able to use the _Read_ operation to fetch more data from the database and then use that 🙂

**Reply by abdonrd** [reply]
Mmmm and how can I select the information?

For example: Get all the Matches from the selected Session.

```json
{
  "filter": {
    "season": {
      "_eq": "ID FROM THE SELECTED SEASON"
    }
  }
}
```

**Reply by rijkvanzanten** [reply]
The selected seasons should be an array of primary keys, so I think you should be able to do something like

```
{
  "filter": {
    "season": {
      "_in": "$trigger.payload.keys"
    }
  }
}
```

(not sure if that's the correct names under `$trigger`)

**Reply by abdonrd** [reply]
Looks like `$trigger.payload.keys` is `undefined`:

```json
{
  "filter": {
    "season": {
      "_in": "{{$trigger.payload.keys}}"
    }
  }
}
```

**Reply by abdonrd** [reply]
Better with `$trigger.body.keys`:

```json
{
  "filter": {
    "season": {
      "_in": "{{$trigger.body.keys}}"
    }
  }
}
```

But:
<img width="623" alt="Screen Shot 2022-08-11 at 18 55 02" src="https://user-images.githubusercontent.com/1007051/184190616-4bf80064-042f-404c-91cc-76bd4c3ffa25.png">

It sends the array as string...

**Reply by rijkvanzanten** [reply]
Ahh yes.. We've seen this problem crop up a couple times. @nickrum I think we might wanna do that trick you proposed earlier where we cast `"{{ json_object_or_array }}"` to actual JSON rather than stringify everything all the time 🤔

**Reply by nickrum** [reply]
Interesting, I though we were already doing that in the initial Flows PR 🤔 Was this changed at some point?

**Reply by rijkvanzanten** [reply]
> Interesting, I though we were already doing that in the initial Flows PR 🤔 Was this changed at some point?

Not that I'm aware of! I remember us talking about it a lot back then too, but don't remember exactly what we end up shipping with.

**Reply by Kamalito75** [reply]
> Ahh yes.. We've seen this problem crop up a couple times. @nickrum I think we might wanna do that trick you proposed earlier where we cast `"{{ json_object_or_array }}` to actual JSON rather than stringify everything all the time 🤔

Hi Rijk. I have the same problem. Is there a workaround please? Maybe using a Transform operation to return JSON rather than a 'stringlified' result?

**Reply by rijkvanzanten** [reply]
@Kamalito75 We just pushed up a tweak in v9.17.0 with that workaround included: https://github.com/directus/directus/pull/15354 . `"{{ val }}"` values are no longer stringified
